File: Snippets/Reverse/rev.lua
--
-- Given a string 'str', return its reverse
--

function reverse (str)
    return (str : reverse ())
end


--
-- Given a number 'num', return its reverse
--

function reverse (num)
    local rev = 0
    while num > 0 do
        rev = rev * 10
        rev = rev + (num % 10)
        num = math . floor (num / 10)
    end
    return (rev)
end

File: Snippets/Reverse/rev.rb
#
# Given a string 'str', return its reverse
#
def reverse (str)
    return (str . reverse)
end

#
# Given a number 'num', return its reverse
#
def reverse (num)
    rev = 0
    while num > 0
        rev *= 10
        rev += num % 10
        num /= 10
    end
    return rev
end
